lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -892,22 +1284,13 @@ def upload_reads(self, ctx, params):
         file_valid = self.validateFASTQ({}, [{'file_path': actualpath,
                                         'interleaved': interleaved}])
         if not file_valid[0][0]['validated']:
-            validation_error_message = "Invalid FASTQ file - Path: " + actualpath + "."
-            if shock:
-                if revid:
-                    validation_error_message += (
-                        " Input Shock IDs - FWD Shock ID : " +
-                        fwdid + ", REV Shock ID : " + revid +
-                        ". FWD File Name : " + fwdname +
-                        ". REV File Name : " + revname +
-                        ". FWD Path : " + fwdpath +
-                        ". REV Path : " + revpath + ".")
-                else:
-                    validation_error_message += (" Input Shock ID : " + fwdid +
-                                                 ". File Name : " + fwdname + ".")
-            elif revpath:
-                validation_error_message += (" Input Files Paths - FWD Path : " +
-                                             fwdpath + ", REV Path : " + revpath + ".")
+            validation_error_message = self._generate_validation_error_message(
```

Comment:
rename it to fwd_location or something - the point is that the combination of fwdid, revid, and reads_source tell you where to get a set of reads - think of it like an object where the api is fwdid and revid and it has different implementations based on the source, which is specified by reads_source. That approach makes the code way less complicated and results in many fewer variables.